File: 28--88-MPH/IronTunes/IronTunes/Song.m
//
//  Song.m
//  IronTunes
//
//  Created by Pedro Trujillo on 11/11/15.
//  Copyright © 2015 Pedro Trujillo. All rights reserved.
//

#import "Song.h"

@implementation Song


- (instancetype)initWithTitle:(NSString * )title artist:(NSString *)artist fileName:(NSString * )filename andAlbumArtWork:(NSString * )albumArtWorkName
{
    if (self = [super init])
    {
        _title = title;
        _artist = artist;
        _filename = filename;
        _albumArtworkName = albumArtWorkName;
        
        //_playerItem = [[AVPlayerItem alloc] initWithURL:[[NSURL alloc] initWithString:[[NSBundle mainBundle] pathForResource:filename ofType:@"mp3"] ]];
        
        _playerItem = [AVPlayerItem playerItemWithURL:[NSURL fileURLWithPath:[[NSBundle mainBundle]pathForResource:filename ofType:@"mp3"]]];
    }
    return self;
}

@end

File: 13--The-Last-Playlist/Iron-Tunes/Iron-Tunes/Song.swift
//
//  Song.swift
//  Iron-Tunes
//
//  Created by Pedro Trujillo on 10/21/15.
//  Copyright © 2015 Pedro Trujillo. All rights reserved.
//

import Foundation
import AVFoundation


class Song
{
    let title: String
    let artist: String
    let fileName: String
    let albumArtworkName : String
    
    let playerItem: AVPlayerItem
    
    
    init(title:String, artist:String, fileName:String, albumArtworkName: String)
    {
        self.title = title
        self.artist = artist
        self.fileName = fileName
        self.albumArtworkName = albumArtworkName
        
        self.playerItem = AVPlayerItem(URL: NSURL(fileURLWithPath: NSBundle.mainBundle().pathForResource(self.fileName, ofType: "mp3")!))
    }
}